app/grandchallenge/reader_studies/forms.py

```diff
@@ -500,6 +525,82 @@ def __init__(self, *args, instance=None, **kwargs):
                         ),
                         initial=val,
                         required=False,
+                        widget=SelectUploadWidget(
+                            attrs={
+                                "reader_study_slug": instance.reader_study.slug,
+                                "display_set_pk": instance.pk,
+                                "interface_pk": interface.pk,
+                            }
+                        ),
                     )
 
             self.fields["order"] = IntegerField(initial=instance.order)
+
+
+class FileForm(Form):
+
+    user_upload = ModelChoiceField(
+        widget=UserUploadSingleWidget,
+        label="File",
+        queryset=None,
+    )
+
+    def __init__(
+        self, *args, user, display_set, interface, instance=None, **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+
+        self.fields["user_upload"].queryset = get_objects_for_user(
+            user, "uploads.change_userupload", accept_global_perms=False
+        ).filter(status=UserUpload.StatusChoices.COMPLETED)
+        self.interface = interface
+        self.display_set = display_set
+
+    def save(self):
+        civ = self.display_set.values.get(interface=self.interface)
```

Comment:
Could there ever be more than one CIV of a certain interface in a display set, in some sort of inbetween state when a previous user upload has not finished yet?

Reply:
That shouldn't be possible as we always remove existing civs for an interface before assigning a new one.